onnxruntime/core/providers/cpu/tensor/onehot.cc

```diff
@@ -158,6 +158,10 @@ Status OneHotOp<in_type, out_type, depth_type>::Compute(OpKernelContext* p_op_ke
   const auto* values_data = values->Data<out_type>();
   Tensor* output = p_op_kernel_context->Output(0, TensorShape(output_shape));
 
+  // edge case where we have a dim with a value of 0
+  if (output->Shape().Size() == 0)
```

Comment:
this seems odd... not the check but the fact that we have to check this...In what scenario would this be true? wouldn't it make more sense to enforce that size > 0

Reply:
If you have a model with a NonZero op you can get a dim value of 0 that flows through all the downstream nodes. They still need to produce an output with a valid shape (i.e. the rank must be correct), but the data will be empty.